kordac/tests/SmokeTests.py

```diff
@@ -2,44 +2,41 @@
 from kordac import Kordac
 
 class SmokeDocsTest(unittest.TestCase):
-    """Tests that docs build if they are found."""
+    '''Tests that docs build if they are found.'''
 
     def __init__(self, *args, **kwargs):
         unittest.TestCase.__init__(self, *args, **kwargs)
         self.maxDiff = None
         self.build_path = "docs/build"
 
-    def setUp(self):
-        pass
-
-    def tearDown(self):
-        pass
-
-    @unittest.skipIf(
-        not os.path.isdir("docs") and os.name not in ['nt', 'posix'],
-        "Docs are not present")
+    @unittest.skipIf(not os.path.isdir('docs') and os.name not in ['nt', 'posix'], 'Docs are not present')
     def test_compile_docs(self):
-        system = os.name
-        # TODO: Cleanup old build
-        command = None
-        if system == 'nt':
-            command = 'make.bat'
-        elif system == 'posix':
-            command = 'make'
-
-        if command is None:
-            self.fail("Unknown operating system, but test still run. This should never happen.")
-
-        p = subprocess.Popen([command, 'clean'], cwd='docs', stdin=subprocess.DEVNULL, stdout=subprocess.DEVNULL, stderr=None)
+        '''This test is skipped if the docs directory is not found.
+        '''
+        cwd = os.path.abspath('docs')
+        command = 'make'
+        options = 'SPHINXOPTS=\'-W\''
+
+        p = subprocess.Popen([command, 'clean'],
+                             cwd=cwd,
+                             stdin=subprocess.DEVNULL,
+                             stdout=subprocess.DEVNULL,
+                             stderr=None,
+                             shell=True)
         p.wait(timeout=10) # Will throw exception if times out
         self.assertEqual(p.returncode, 0) # Success returncode
 
-        p = subprocess.Popen([command, 'SPHINXOPTS=\'-W\'', 'html'], cwd='docs', stdin=subprocess.DEVNULL, stdout=subprocess.DEVNULL, stderr=None)
+        p = subprocess.Popen([command, options, 'html'],
+                             cwd=cwd,
+                             stdin=subprocess.DEVNULL,
+                             stdout=subprocess.DEVNULL,
+                             stderr=None,
+                             shell=True)
         p.wait(timeout=10) # Will throw exception if times out
```

Comment:
We essentially kickoff a command in the system shell (bash for linux) with the given working directory, and all the input and output streams set to the os null file (as we do not careabout the i/o BUT leave the error stream to the current python error stream. 

We then wait up to 10 seconds for the command to complete, if it takes over 10 seconds an exception is raised meaning the test fails.

If the command did complete we check its exit code, where 0 is considered a success.